### ISSUE TITLE
Are we taking right approach with junction lane routing UI overhaul?

### ISSUE THREAD

**Issue by aubergine10** [krzychu124/Cities-Skylines-Traffic-Manager-President-Edition#481]
> **Background:** I used to play a game called [Carmageddon])(https://en.wikipedia.org/wiki/Carmageddon) at a friends house, then drive home afterwards while my brain was still in "collide with as many pedestrians as possible" mode. These days, it's C:SL and my brain is in "focus on the efficiency of the transit infrastructure" mode.

In C:SL we, as users, work from the 'mayor perspective' - looking from above and our focus is on getting that traffic flowing smoothly.

The existing TM:PE UI, however, is often done from 'driver perspective' - the view from being in a car on the road. That's why some of it (lane arrows and timed traffic lights in particular) are so cumbersome to use.

We definitely want UI interactions to work with the mayor perspective, but sometimes that disconnects with the driver perspective. How do we connect the two perspectives where necessary?

I'll use lane arrows as an illustration...

### Lane arrows

We are overhauling this UI from driver perspective to mayor perspective. That's necessary, as it will make it easier to use, but this feature in particular has a big disconnect between the two perspectives:

* Mayor: _If you enter via this lane, you can exit via these segments_
* Driver: _I can exit via any segments in the general direction of the arrow_

I think by a taking a mayor-only approach to the UI we've blurred the lines too much between lane _arrows_ and lane _connectors_, and we've also omitted any sense of the driver perspective. As a result, I'm thinking we might be harming gameplay too.

Here's what I propose...

1 Select incoming lane
2. Click an outgoing lane
3. An arc visualises the 'general direction' (lane arrow direction)
4. All outgoing segments in that direction are treated as being selected

Hastily thrown together mockup - diver is entering from `0` and mayor has clicked outgoing segment `1`:

![image](https://user-images.githubusercontent.com/1386719/62845559-f021cb00-bcc0-11e9-99ac-ff3ad6f6563e.png)

So the UI is still mayor perspective, but the arc shows driver perspective which is what ultimately decides the valid exit segments for that incoming lane.

This will also very clearly differentiate between lane arrows and lane connectors.

Lane connectors is more like painting dashed lines over the junction to more accurately control traffic flow:

![image](https://user-images.githubusercontent.com/1386719/62845720-53f8c380-bcc2-11e9-83c2-ce542a0e34c9.png)

Tagging #391

**Comment by kvakvs** [comment]
With my existing prototype (outdated now because master changed so much) it means i will need to show less lane curves (hey, they looked cool!) and add arcs for directions instead. This, in fact, might be a nice improvement to visualize what each arrow direction means.

**Comment by kvakvs** [comment]
Wait @aubergine10 do you also want an incoming arc? The thin light blue one. Selecting all lanes in the arc will defeat lane fine-tuning ability, where some lanes on some roads can be sent in different directions.

**Comment by aubergine10** [comment]
No, that was just an artifact of the graph I butchered to create a mockup image hehehe

Just going in game to get some screenshots

**Comment by aubergine10** [comment]
Also, I think we'd still need 'lane connector' style lines from incoming lanes to outgoing lanes, in particular so user can hover over nearby incoming lanes to see what they connect to. Also, hovering outgoing lanes should show which incoming lanes connect to them if possible?

Those lines serve another purpose too. For example, if I have a crossroads junction with all 4 lane 2-way roads, it would be good to show what the lanes are connecting to as it might not be immediately obvious from the arrows or direction arcs. 

Rough mockup:

![image](https://user-images.githubusercontent.com/1386719/63211146-55bfee00-c0eb-11e9-936f-8746e2daf8c0.png)

So if only one incoming lane makes the turn in given direction, it might lead to two outgoing lanes (I have no idea how this currently works - because there's no visualisation of it in-game). If two lanes go in a direction, then the outgoing lanes are sort of spread between them.

@VictorPhilipp did a good page explaining what happens - using lane connectors to illustrate his points hehe: https://tmpe.viathinksoft.com/wiki/index.php?title=Junction_%26_transition_routing
